server/src/main/java/org/elasticsearch/search/profile/Profilers.java

```diff
@@ -16,45 +16,28 @@
 import org.elasticsearch.search.profile.query.QueryProfileShardResult;
 import org.elasticsearch.search.profile.query.QueryProfiler;
 
-import java.util.ArrayList;
 import java.util.Collections;
-import java.util.List;
 
 /** Wrapper around all the profilers that makes management easier. */
 public final class Profilers {
 
     private final ContextIndexSearcher searcher;
-    private final List<QueryProfiler> queryProfilers = new ArrayList<>();
+    private final QueryProfiler queryProfiler;
     private final AggregationProfiler aggProfiler = new AggregationProfiler();
     private DfsProfiler dfsProfiler;
 
     public Profilers(ContextIndexSearcher searcher) {
         this.searcher = searcher;
```

Comment:
I believe you no longer need to keep a reference to the searcher if you call searcher#setProfiler in the constructor